File: src/AddIns/Misc/PackageManagement/Project/Src/EnvDTE/Document.cs
// Copyright (c) AlphaSierraPapa for the SharpDevelop Team (for details please see \doc\copyright.txt)
// This code is distributed under the GNU LGPL (for details please see \doc\license.txt)

using System;
using ICSharpCode.SharpDevelop.Gui;

namespace ICSharpCode.PackageManagement.EnvDTE
{
	public class Document : MarshalByRefObject, global::EnvDTE.Document
	{
		IViewContent view;
		
		public Document(string fileName, IViewContent view)
		{
			this.FullName = fileName;
			this.view = view;
		}
		
		public virtual bool Saved {
			get { return !view.IsDirty; }
			set { view.PrimaryFile.IsDirty = !value; }
		}
		
		public string FullName { get; private set; }
		
		public Object Object(string modelKind)
		{
			throw new NotImplementedException();
		}
	}
}

File: src/AddIns/Misc/PackageManagement/SharpDevelop.EnvDTE/Src/Document.vb
' Copyright (c) AlphaSierraPapa for the SharpDevelop Team (for details please see \doc\copyright.txt)
' This code is distributed under the GNU LGPL (for details please see \doc\license.txt)

Namespace EnvDTE
	Public Interface Document
		Property Saved() As Boolean
		ReadOnly Property FullName() As String
		
		Function [Object](modelKind As String) As Object
	End Interface
End Namespace